spec/rspec/core/option_parser_spec.rb

```diff
@@ -81,28 +83,101 @@ def generate_help_text
 
     %w[ -v --version ].each do |option|
       describe option do
-        it "prints the version and exits" do
+        it "prints the version and returns a zero exit code" do
           parser = Parser.new([option])
-          expect(parser).to receive(:exit)
 
           expect {
-            parser.parse
+            options = parser.parse
+            exit_code = options[:runner].call
+            expect(exit_code).to eq(0)
           }.to output("#{RSpec::Core::Version::STRING}\n").to_stdout
         end
       end
     end
 
+    %w[ -X --drb ].each do |option|
+      describe option do
+        let(:parser) { Parser.new([option]) }
+        let(:options) { double(:options) }
+        let(:err) { double(:stderr, :puts => nil) }
+        let(:out) { double(:stderr, :puts => nil) }
```

Comment:
Actually, the interface for IO objects is quite large (making doubles that stand-in for an IO pretty brittle) and ruby provides a perfectly usable double that _exactly_ mimics the full interface in `StringIO` -- so I think I'd prefer to see a `StringIO` used here instead.